### PR TITLE
 Document building and testing

### PR DESCRIPTION
**Pull request by gregkare**
Also remove the Makefile, move the test command to a script and add a doc npm script

Connected to #1016

### CODE REVIEW

File: doc/contributing/build-commands.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+   $ npm run dev
+
+doc
+---
+
+Generates the documentation website. See :doc:`the Documentation section of
+the Contributing docs</contributing/docs>` to install the dependencies::
+
+   $ npm run doc
```
Comment (skddc) [inline doc/contributing/build-commands.rst]: This belongs in the documentation howto, not here.

**Comment by skddc** [comment]
Reg. the "build commands" doc: it's more of an annotated but incomplete list of npm scripts than it is a "how to build and test the library" howto for new contributors. E.g. the test command is not alone by itself, but the important info is to also explain how to test single files, because you hardly want to wait for like 90 seconds for the tests to complete after every code change. And the dev command should ideally explain how to use `npm link` for testing live code changes in an app.

**Comment by gregkare** [comment]
I will explain how to use `npm link` next

**Comment by skddc** [comment]
The point was actually that the structure of the document doesn't make sense like this.

**Comment by galfert** [comment]
The `.travis.yml` also needs to be updated, it's still using `make` commands.

**Comment by skddc** [comment]
I noticed that we don't currently run JSHint in the test suite and on CI, and the config is also a bit outdated. Will create a seperate issue and just pretend it's there for the new doc I just added to this PR.

**Comment by skddc** [comment]
I took the liberty to finish this one. Ready for review/merge.

**Comment by gregkare** [comment]
LGTM!

File: doc/contributing/testing.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   If you add ``./node_modules/.bin`` to your ``PATH``, you can call
+   executables in any npm project directly. For example in ``~/.bashrc``, add
+   the line ``export PATH=$PATH:./node_modules/.bin`` (and run ``source
+   ~/.bashrc`` to load that change in open terminal sessions).  Then you can
+   just run ``jaribu test/unit/foo_suite.js``.
+
+Continous integration
+---------------------
+
+The rs.js test suite is by Travis CI on every push to our repo on GitHub.  When
```
Comment (galfert) [inline doc/contributing/testing.rst]: Missing word: "test suite is **run** by Travis CI"

Comment (skddc) [reply doc/contributing/testing.rst]: Thanks!

**Pull request merged**